lib/ansible/modules/cloud/amazon/s3.py

```diff
@@ -706,41 +774,46 @@ def main():
     module.exit_json(failed=False)
 
 
-def get_s3_connection(aws_connect_kwargs, location, rgw, s3_url):
+def get_s3_connection(module, aws_connect_kwargs, location, rgw, s3_url):
     if s3_url and rgw:
         rgw = urlparse(s3_url)
-        s3 = boto.connect_s3(
-            is_secure=rgw.scheme == 'https',
-            host=rgw.hostname,
-            port=rgw.port,
-            calling_format=OrdinaryCallingFormat(),
-            **aws_connect_kwargs
-        )
+        s3 = boto3_conn(module,
+                        conn_type='client',
+                        resource='s3',
+                        region=location,
+                        endpoint=s3_url,
+                        **aws_connect_kwargs)
     elif is_fakes3(s3_url):
+        aws_connect_kwargs["region"] = 'us-gov-west-1'
         fakes3 = urlparse(s3_url)
-        s3 = S3Connection(
-            is_secure=fakes3.scheme == 'fakes3s',
-            host=fakes3.hostname,
-            port=fakes3.port,
-            calling_format=OrdinaryCallingFormat(),
-            **aws_connect_kwargs
+        boto3.resource.meta.client.meta.events.unregister('before-sign.s3', fix_s3_host)
+        session = boto3.session.Session(**aws_connect_kwargs)
+        s3 = session.client(service_name='s3',
+                            endpoint_url=fakes3,
+                            use_ssl=fakes3.scheme == 'fakes3s',
+                            host=fakes3.hostname,
+                            port=fakes3.port,
+                            **aws_connect_kwargs
         )
     elif is_walrus(s3_url):
         walrus = urlparse(s3_url).hostname
-        s3 = boto.connect_walrus(walrus, **aws_connect_kwargs)
+        s3 = boto3_conn(module,
+                        conn_type='client',
+                        resource='s3',
+                        region=location,
+                        endpoint=walrus
+                        **aws_connect_kwargs
+        )
+                        
```

Comment:
CI failure due to PEP 8 issue:

```
2017-02-27 20:21:42 ERROR: PEP 8: lib/ansible/modules/cloud/amazon/s3.py:807:1: W293 blank line contains whitespace (current)
```

The PEP 8 tests can be run locally with `make pep8`.